CactbotOverlay/CactbotOverlay.cs

```diff
@@ -247,6 +253,79 @@ public CactbotOverlay(CactbotOverlayConfig config)
       fast_update_timer_semaphore_.Release();
     }
 
+    public void EnableWatchers(bool check) {
+      this.watcher_state = check;
+    foreach (var watcher in this.watchers)
+        watcher.EnableRaisingEvents = check;
+    }
+ 
+    private void SetupFileWatcher(string url) {
+      if (url == "")
+        return;
+
+      foreach(var watcher in this.watchers)
+        watcher.Dispose();
+
+      List<string> path_list = new List<string>();
+      var overlay_full_path = url;
+      string base_path = null;
+
+      if (!url.Contains("://quisquous.github.io/cactbot/")) {
+        // Path of overlay if overlay is not remote!
+        overlay_full_path = Regex.Replace(overlay_full_path, @"file:[\\\/]+", "");
+        var overlay_path = Path.GetDirectoryName(url);
+        overlay_path = Regex.Replace(overlay_path, @"file:[\\\/]+", "");
+        base_path = Regex.Match(overlay_path, @"(.*cactbot)").Groups[1].ToString();
+
+        path_list.Add(base_path);
+
+        // Path of HTML resources if overlay is not remote!
+        string html = File.ReadAllText(overlay_full_path);
+        var html_path = Regex.Match(html, @"(?:src|href)=""(.*)[\\\/].*\..*""");
+        foreach(var match in html_path.Groups) {
+          string match_path = match.ToString();
+          if (match_path.StartsWith(".."))
+            match_path = overlay_path + "/" + match_path;
+          if (!match_path.Contains(base_path) && Directory.Exists(match_path) && !path_list.Contains(match_path))
+            path_list.Add(match_path);
+        }
+      }
+
+      // User Dir
+      string user_path;
+      Dictionary<string, string> local_UserFiles = new Dictionary<string, string>();
+      GetUserConfigDirAndFiles(out user_path, out local_UserFiles);
```

Comment:
This is pretty unfortunate, as it causes all the user files to get loaded twice.  Is there any way around that?

Reply:
Since the `user_path` is not assigned to an accessible variable, I guess not?